test_defopt.py

```diff
@@ -78,7 +82,7 @@ def bad(**kwargs):
             defopt.run(bad)
 
     def test_no_function(self):
-        with self.assertRaisesRegex(ValueError, 'at least one'):
+        with self.assertRaises(TypeError):
             defopt.run()
```

Comment:
The analogue of what this was previously testing for would be `defopt.run([])`, which currently produces a useless parser. I think we should continue to reject this.